pkgs/by-name/db/dbvisualizer/package.nix

```diff
@@ -0,0 +1,73 @@
+{
+  lib,
+  copyDesktopItems,
+  fetchurl,
+  makeDesktopItem,
+  makeWrapper,
+  stdenv,
+  temurin-jre-bin-17,
```

Comment:
Does it *have* to be Temurin? (i.e. does `openjdk17` work?)

Reply:
It does not need to be temurin. My reasoning was that `temurin-jre-bin-17` is 136MB as opposed to `openjdk17` which is 470MB. Makes sense, since it is just the runtime as opposed to an sdk.

Probably there is a way to take only the jre from openjdk? When looking through nixpkgs I only found `jre17_minimal` as a potential candidate, but I was not sure what `minimal` implied. And sure enough, it does not run with `jre17_minimal`. Thus I just went for temurin.

Reply:
We removed the JRE-only builds from Nixpkgs a while back as a part of an overall refactor. OpenJDK is preferred since it's something that we built ourselves, rather than Temurin which is a prebuilt, binary-only distribution.

Reply:
After doing some more digging, I think you might be able to make `jre_minimal` work by overriding it with more modules, like `jre17_minimal.override { modules = [ "java.base" "java.logging" ]; }`. Yes, this is undocumented, yes, this sucks, I'll probably make a separate issue about it

Reply:
I have tried using this, however it did not even open. after adding some more modules, I got it to open at least a graphical window, but it would still give an error within the window.

I ended up using `jre17_minimal.override { modules = [ "java.base" "java.logging" "java.desktop" "java.xml" "java.sql" ]; }`, which got it to actually present a graphical window. It still gives the following internal error: `java.lang.NoClassDefFoundError: javax/management/NotificationListener`.

Point is: I do not think it is worth fiddling with this, because chances are I am going to miss adding at least one important java module. I do not think this is worth it.